--- crypto/x509/by_dir.c
+++ crypto/x509/by_dir.c
@@ -305,7 +305,12 @@ get_cert_by_subject(X509_LOOKUP *xl, int type, X509_NAME *name,
 
 	ctx = (BY_DIR *)xl->method_data;
 
-	h = X509_NAME_hash(name);
+	#if defined(__ANDROID__)
+		// See https://github.com/openssl/openssl/issues/13565#issuecomment-812768324
+		h = X509_NAME_hash_old(name);
+	#else
+		h = X509_NAME_hash(name);
+	#endif
 	for (i = 0; i < sk_BY_DIR_ENTRY_num(ctx->dirs); i++) {
 		BY_DIR_ENTRY *ent;
 		int idx;
@@ -420,3 +425,4 @@ finish:
 		BUF_MEM_free(b);
 	return (ok);
 }
+

